lib/base/io-engine.hpp

```diff
@@ -102,21 +107,32 @@ class IoEngine
 	static void SpawnCoroutine(Handler& h, Function f) {
 
 		boost::asio::spawn(h,
+#if BOOST_VERSION >= 108000
+			std::allocator_arg_t(),
+			boost::context::fixedsize_stack(GetCoroutineStackSize()),
+#endif // BOOST_VERSION >= 108000
 			[f](boost::asio::yield_context yc) {
 
 				try {
 					f(yc);
-				} catch (const boost::coroutines::detail::forced_unwind &) {
-					// Required for proper stack unwinding when coroutines are destroyed.
-					// https://github.com/boostorg/coroutine/issues/39
-					throw;
 				} catch (const std::exception& ex) {
 					Log(LogCritical, "IoEngine") << "Exception in coroutine: " << DiagnosticInformation(ex);
 				} catch (...) {
-					Log(LogCritical, "IoEngine", "Exception in coroutine!");
+					try {
+						Log(LogCritical, "IoEngine", "Exception in coroutine!");
+					} catch (...) {
+					}
+
+					// Required for proper stack unwinding when coroutines are destroyed.
+					// https://github.com/boostorg/coroutine/issues/39
+					throw;
 				}
 			},
+#if BOOST_VERSION >= 108000
+			boost::asio::detached
+#else // BOOST_VERSION >= 108000
 			boost::coroutines::attributes(GetCoroutineStackSize()) // Set a pre-defined stack size.
+#endif // BOOST_VERSION >= 108000
```

Comment:
This also affects:

* Fedora 39+ (all)
* Ubuntu 24.04+

Reply:
I don't understand what you are trying to say! The check from https://github.com/Icinga/icinga2/pull/10278#pullrequestreview-2542879016 was actually exactly how I expected it to be, but you just changed it without any explanation and posted this afterwards. Isn't this all about boost `>= 1.87`, so why change the implementation for all boost `1.80+` versions?